File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/SplitSettlement.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class SplitSettlement : ISerialized
    {
        public SplitSettlement(XmlNode xmlNode)
        {
            XmlNode splitSettlementAmountNode = xmlNode.SelectSingleNode("splitSettlementAmount");
            
            if (splitSettlementAmountNode != null)
            {
                if (splitSettlementAmountNode.Attributes["href"] != null || splitSettlementAmountNode.Attributes["id"] != null) 
                {
                    if (splitSettlementAmountNode.Attributes["id"] != null) 
                    {
                        splitSettlementAmountIDRef_ = splitSettlementAmountNode.Attributes["id"].Value;
                        Money ob = new Money(splitSettlementAmountNode);
                        IDManager.SetID(splitSettlementAmountIDRef_, ob);
                    }
                    else if (splitSettlementAmountNode.Attributes["href"] != null)
                    {
                        splitSettlementAmountIDRef_ = splitSettlementAmountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        splitSettlementAmount_ = new Money(splitSettlementAmountNode);
                    }
                }
                else
                {
                    splitSettlementAmount_ = new Money(splitSettlementAmountNode);
                }
            }
            
        
            XmlNode beneficiaryBankNode = xmlNode.SelectSingleNode("beneficiaryBank");
            
            if (beneficiaryBankNode != null)
            {
                if (beneficiaryBankNode.Attributes["href"] != null || beneficiaryBankNode.Attributes["id"] != null) 
                {
                    if (beneficiaryBankNode.Attributes["id"] != null) 
                    {
                        beneficiaryBankIDRef_ = beneficiaryBankNode.Attributes["id"].Value;
                        Routing ob = new Routing(beneficiaryBankNode);
                        IDManager.SetID(beneficiaryBankIDRef_, ob);
                    }
                    else if (beneficiaryBankNode.Attributes["href"] != null)
                    {
                        beneficiaryBankIDRef_ = beneficiaryBankNode.Attributes["href"].Value;
                    }
                    else
                    {
                        beneficiaryBank_ = new Routing(beneficiaryBankNode);
                    }
                }
                else
                {
                    beneficiaryBank_ = new Routing(beneficiaryBankNode);
                }
            }
            
        
            XmlNode beneficiaryNode = xmlNode.SelectSingleNode("beneficiary");
            
            if (beneficiaryNode != null)
            {
                if (beneficiaryNode.Attributes["href"] != null || beneficiaryNode.Attributes["id"] != null) 
                {
                    if (beneficiaryNode.Attributes["id"] != null) 
                    {
                        beneficiaryIDRef_ = beneficiaryNode.Attributes["id"].Value;
                        Routing ob = new Routing(beneficiaryNode);
                        IDManager.SetID(beneficiaryIDRef_, ob);
                    }
                    else if (beneficiaryNode.Attributes["href"] != null)
                    {
                        beneficiaryIDRef_ = beneficiaryNode.Attributes["href"].Value;
                    }
                    else
                    {
                        beneficiary_ = new Routing(beneficiaryNode);
                    }
                }
                else
                {
                    beneficiary_ = new Routing(beneficiaryNode);
                }
            }
            
        
        }
        
    
        #region SplitSettlementAmount_
        private Money splitSettlementAmount_;
        public Money SplitSettlementAmount_
        {
            get
            {
                if (this.splitSettlementAmount_ != null)
                {
                    return this.splitSettlementAmount_; 
                }
                else if (this.splitSettlementAmountIDRef_ != null)
                {
                    splitSettlementAmount_ = IDManager.getID(splitSettlementAmountIDRef_) as Money;
                    return this.splitSettlementAmount_; 
                }
                else
                {
                      return this.splitSettlementAmount_; 
                }
            }
            set
            {
                if (this.splitSettlementAmount_ != value)
                {
                    this.splitSettlementAmount_ = value;
                }
            }
        }
        #endregion
        
        public string splitSettlementAmountIDRef_ { get; set; }
        #region BeneficiaryBank_
        private Routing beneficiaryBank_;
        public Routing BeneficiaryBank_
        {
            get
            {
                if (this.beneficiaryBank_ != null)
                {
                    return this.beneficiaryBank_; 
                }
                else if (this.beneficiaryBankIDRef_ != null)
                {
                    beneficiaryBank_ = IDManager.getID(beneficiaryBankIDRef_) as Routing;
                    return this.beneficiaryBank_; 
                }
                else
                {
                      return this.beneficiaryBank_; 
                }
            }
            set
            {
                if (this.beneficiaryBank_ != value)
                {
                    this.beneficiaryBank_ = value;
                }
            }
        }
        #endregion
        
        public string beneficiaryBankIDRef_ { get; set; }
        #region Beneficiary_
        private Routing beneficiary_;
        public Routing Beneficiary_
        {
            get
            {
                if (this.beneficiary_ != null)
                {
                    return this.beneficiary_; 
                }
                else if (this.beneficiaryIDRef_ != null)
                {
                    beneficiary_ = IDManager.getID(beneficiaryIDRef_) as Routing;
                    return this.beneficiary_; 
                }
                else
                {
                      return this.beneficiary_; 
                }
            }
            set
            {
                if (this.beneficiary_ != value)
                {
                    this.beneficiary_ = value;
                }
            }
        }
        #endregion
        
        public string beneficiaryIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/SplitSettlement.cpp
// SplitSettlement.cpp 
#include "SplitSettlement.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

SplitSettlement::SplitSettlement(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //splitSettlementAmountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* splitSettlementAmountNode = xmlNode->FirstChildElement("splitSettlementAmount");

   if(splitSettlementAmountNode){splitSettlementAmountIsNull_ = false;}
   else{splitSettlementAmountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- splitSettlementAmountNode , address : " << splitSettlementAmountNode << std::endl;
   #endif
   if(splitSettlementAmountNode)
   {
      if(splitSettlementAmountNode->Attribute("href") || splitSettlementAmountNode->Attribute("id"))
      {
          if(splitSettlementAmountNode->Attribute("id"))
          {
             splitSettlementAmountIDRef_ = splitSettlementAmountNode->Attribute("id");
             splitSettlementAmount_ = boost::shared_ptr<Money>(new Money(splitSettlementAmountNode));
             splitSettlementAmount_->setID(splitSettlementAmountIDRef_);
             IDManager::instance().setID(splitSettlementAmountIDRef_,splitSettlementAmount_);
          }
          else if(splitSettlementAmountNode->Attribute("href")) { splitSettlementAmountIDRef_ = splitSettlementAmountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { splitSettlementAmount_ = boost::shared_ptr<Money>(new Money(splitSettlementAmountNode));}
   }

   //beneficiaryBankNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* beneficiaryBankNode = xmlNode->FirstChildElement("beneficiaryBank");

   if(beneficiaryBankNode){beneficiaryBankIsNull_ = false;}
   else{beneficiaryBankIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- beneficiaryBankNode , address : " << beneficiaryBankNode << std::endl;
   #endif
   if(beneficiaryBankNode)
   {
      if(beneficiaryBankNode->Attribute("href") || beneficiaryBankNode->Attribute("id"))
      {
          if(beneficiaryBankNode->Attribute("id"))
          {
             beneficiaryBankIDRef_ = beneficiaryBankNode->Attribute("id");
             beneficiaryBank_ = boost::shared_ptr<Routing>(new Routing(beneficiaryBankNode));
             beneficiaryBank_->setID(beneficiaryBankIDRef_);
             IDManager::instance().setID(beneficiaryBankIDRef_,beneficiaryBank_);
          }
          else if(beneficiaryBankNode->Attribute("href")) { beneficiaryBankIDRef_ = beneficiaryBankNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { beneficiaryBank_ = boost::shared_ptr<Routing>(new Routing(beneficiaryBankNode));}
   }

   //beneficiaryNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* beneficiaryNode = xmlNode->FirstChildElement("beneficiary");

   if(beneficiaryNode){beneficiaryIsNull_ = false;}
   else{beneficiaryIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- beneficiaryNode , address : " << beneficiaryNode << std::endl;
   #endif
   if(beneficiaryNode)
   {
      if(beneficiaryNode->Attribute("href") || beneficiaryNode->Attribute("id"))
      {
          if(beneficiaryNode->Attribute("id"))
          {
             beneficiaryIDRef_ = beneficiaryNode->Attribute("id");
             beneficiary_ = boost::shared_ptr<Routing>(new Routing(beneficiaryNode));
             beneficiary_->setID(beneficiaryIDRef_);
             IDManager::instance().setID(beneficiaryIDRef_,beneficiary_);
          }
          else if(beneficiaryNode->Attribute("href")) { beneficiaryIDRef_ = beneficiaryNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { beneficiary_ = boost::shared_ptr<Routing>(new Routing(beneficiaryNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<Money> SplitSettlement::getSplitSettlementAmount()
{
   if(!this->splitSettlementAmountIsNull_){
        if(splitSettlementAmountIDRef_ != ""){
             return boost::shared_static_cast<Money>(IDManager::instance().getID(splitSettlementAmountIDRef_));
        }else{
             return this->splitSettlementAmount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Money>();
   }
}
boost::shared_ptr<Routing> SplitSettlement::getBeneficiaryBank()
{
   if(!this->beneficiaryBankIsNull_){
        if(beneficiaryBankIDRef_ != ""){
             return boost::shared_static_cast<Routing>(IDManager::instance().getID(beneficiaryBankIDRef_));
        }else{
             return this->beneficiaryBank_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Routing>();
   }
}
boost::shared_ptr<Routing> SplitSettlement::getBeneficiary()
{
   if(!this->beneficiaryIsNull_){
        if(beneficiaryIDRef_ != ""){
             return boost::shared_static_cast<Routing>(IDManager::instance().getID(beneficiaryIDRef_));
        }else{
             return this->beneficiary_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Routing>();
   }
}
}

